File: starter-bot/src/main/java/ru/croccode/hypernull/bot/Bot.java
package ru.croccode.hypernull.bot;

import ru.croccode.hypernull.message.Hello;
import ru.croccode.hypernull.message.MatchOver;
import ru.croccode.hypernull.message.MatchStarted;
import ru.croccode.hypernull.message.Move;
import ru.croccode.hypernull.message.Register;
import ru.croccode.hypernull.message.Update;

public interface Bot {

	Register onHello(Hello hello);

	void onMatchStarted(MatchStarted matchStarted);

	Move onUpdate(Update update);

	void onMatchOver(MatchOver matchOver);
}

File: starter-bot-python/bot/bot.py
import random
from dataclasses import dataclass, field

from client.message.extra_types import Mode
from client.message.messages import MatchStarted, Update

bot_name = 'nGragas'
bot_secret = ''
mode = Mode.FRIENDLY


@dataclass
class Bot:
    name: str
    secret: str = None
    mode: Mode = Mode.FRIENDLY
    match_info: MatchStarted = field(init=False)

    def on_match_started(self, match_info: MatchStarted):
        self.match_info = match_info
        print('Match started!')

    def on_update(self, update: Update) -> tuple[int, int]:
        round_number = update.round
        bots = update.bot
        coins = update.coin
        blocks = update.block

        dx = random.choice([-1, 0, 1])
        dy = random.choice([-1, 0, 1])
        return dx, dy

    def on_match_over(self) -> None:
        print('Match over')
